src/ui/components/PrefixBadge.tsx

```diff
@@ -1,52 +1,28 @@
 import { setBreakpointPrefixBadge } from "devtools/client/debugger/src/actions/breakpoints";
 import { PrefixBadge } from "devtools/client/debugger/src/reducers/types";
 import { Breakpoint } from "devtools/client/debugger/src/selectors";
-import React, { useState, useRef, MutableRefObject } from "react";
+import React from "react";
 import { useDispatch } from "react-redux";
 import { useFeature } from "ui/hooks/settings";
-import useModalDismissSignal from "ui/hooks/useModalDismissSignal";
+import { PrefixBadgePicker } from "components";
 
 import styles from "./PrefixBadge.module.css";
 
-type SelectBadge = (prefixBadge?: PrefixBadge) => void;
-
 // Rendered in log point PanelEditor, when a log point is being edited.
 // Clicking this item will show the PrefixBadgePicker.
 export default function PrefixBadgeButton({ breakpoint }: { breakpoint: Breakpoint }) {
-  const [showPrefixBadge, setShowPrefixBadge] = useState<boolean>(false);
   const { value: enableUnicornConsole } = useFeature("unicornConsole");
   const dispatch = useDispatch();
 
   if (!enableUnicornConsole) {
     return null;
   }
 
-  const prefixBadge = breakpoint.options.prefixBadge;
-  const className =
-    prefixBadge === "unicorn"
-      ? `${styles.ColorBadgeButton} ${styles.UnicornBadge}`
-      : prefixBadge != null
-      ? `${styles.ColorBadgeButton} ${styles[prefixBadge]}`
-      : `${styles.ColorBadgeButton} ${styles.ColorBadgeButtonEmpty}`;
-
   return (
-    <div className={styles.PrefixBadgeButton}>
-      <button
-        className={className}
-        onClick={() => {
-          setShowPrefixBadge(!showPrefixBadge);
-        }}
+    <div className="relative z-10" style={{ width: 28, marginLeft: 7 }}>
+      <PrefixBadgePicker
+        onSelect={newPrefixBadge => dispatch(setBreakpointPrefixBadge(breakpoint, newPrefixBadge))}
       />
-      {showPrefixBadge && (
-        <PrefixBadgePicker
-          onDismiss={() => setShowPrefixBadge(false)}
-          onSelect={newPrefixBadge => {
-            console.log("onSelect() newPrefixBadge:", newPrefixBadge);
```

Comment:
Not sure if this was expected to be here, but I removed it since it seemed like leftover code.